src/Pawn.cpp

```diff
@@ -0,0 +1,30 @@
+#include "Pawn.h"
+
+Pawn::Pawn(bool isWhite)
+	:Piece(isWhite)
+{
+}
+
+bool Pawn::isCorrectMove(const Location& source, const Location& destination) const
+{
+    if (m_firstMove)
+        if (this->getColor())
+            return (destination.x == source.x + 2 || destination.x == source.x + 1) && destination.y == source.y;
+        else
+            return (destination.x == source.x - 2 || destination.x == source.x - 1) && destination.y == source.y;
+    else
+        if (this->getColor())
+            return destination.x == source.x + 1 && destination.y == source.y;
+        else

```

Comment:
It is generally recommended to add braces {} for nested if statements to improve code readability and reduce the chances of introducing bugs.